src/System.Management.Automation/engine/Attributes.cs

```diff
@@ -1098,6 +1098,194 @@ private static Type GetCommonType(Type minType, Type maxType)
     }
 
     /// <summary>
+    /// Base class for postive number valididation attributes
+    /// </summary>
+    /// <remarks>
+    /// Base class used by postive and non negative attribute
+    /// <seealso cref="ValidatePositiveAttribute"/>,
+    /// <seealso cref="ValidateNonNegativeAttribute"/>
+    /// </remarks>
+    [AttributeUsage(AttributeTargets.Field | AttributeTargets.Property)]
+    public class ValidatePositiveBaseAttribute : ValidateEnumeratedArgumentsAttribute
+    {
+        /// <summary>
+        /// Minimum allowed value for the attribute
+        /// </summary>
+        public int minValue;
```

Comment:
Any tips on how to include the larger numbers? I was thinking if checking if the element has a ```CompareTo()``` but not sure if this is correct. For example, string also have compare to which would be an invalid type.